edk2toollib/os/uefivariablesupport.py

```diff
@@ -0,0 +1,331 @@
+                    c_void_p,
+                    c_int,
+                ]
+                self._SetFirmwareEnvironmentVariableEx = (
+                    windll.kernel32.SetFirmwareEnvironmentVariableExW
+                )
+                self._SetFirmwareEnvironmentVariableEx.restype = c_int
+                self._SetFirmwareEnvironmentVariableEx.argtypes = [
+                    c_wchar_p,
+                    c_wchar_p,
+                    c_void_p,
+                    c_int,
+                    c_int,
+                ]
+            except Exception:
+                logging.warn(
+                    "G[S]etFirmwareEnvironmentVariableW function doesn't seem to exist"
+                )
+        else:
+            pass
+
+    def GetUefiVar(self, name: str, guid: str) -> tuple[int, str]:
+        """_Retrieve Uefi Variable from the system.
+
+        Args:
+            name (str): String corresponding to the Unicode Name of the variable.
+            guid (str): String corresponding to the Uefi Guid of the variable.
+
+        Returns:
+            Tuple: (error code, string of variable data)
+        """
+        err = 0
+        if os.name == 'nt':
+            efi_var = create_string_buffer(EFI_VAR_MAX_BUFFER_SIZE)
+            if self._GetFirmwareEnvironmentVariable is not None:
+                logging.info(
+                    "calling GetFirmwareEnvironmentVariable( name='%s', GUID='%s' ).."
+                    % (name, "{%s}" % guid)
+                )
+                length = self._GetFirmwareEnvironmentVariable(
+                    name, "{%s}" % guid, efi_var, EFI_VAR_MAX_BUFFER_SIZE
+                )
+            if (0 == length) or (efi_var is None):
+                err = windll.kernel32.GetLastError()
+                if err != 0 and err != UefiVariable.ERROR_ENVVAR_NOT_FOUND:
+                    logging.error(
+                        "GetFirmwareEnvironmentVariable[Ex] failed (GetLastError = 0x%x)" % err
+                    )
+                    logging.error(WinError())
+                if efi_var is None:
+                    return (err, None)
+            return (err, efi_var[:length])
+
+        else:
+            # the variable name is VariableName-Guid
+            path = '/sys/firmware/efi/efivars/' + name + '-%s' % guid
+
+            if not os.path.exists(path):
+                err = UefiVariable.ERROR_ENVVAR_NOT_FOUND
+                return (err, None)
+
+            efi_var = create_string_buffer(EFI_VAR_MAX_BUFFER_SIZE)
+            with open(path, 'rb') as fd:
+                efi_var = fd.read()
+
+            return (err, efi_var[:length])
+
+    def GetUefiAllVarNames(self) -> tuple[int, bytes]:
+        """Get all Uefi Variables in the system, and return a byte packed byte string.
+
+        Raises:
+            Exception: Returned variable could not be parsed.
+
+        Returns:
+            tuple[int, bytes]:
+            Integer is return status, 0 for error, non-zero for success.
```

Comment:
The documentation after `tuple[int, bytes]: needs to be indented. That will fix the ruff error.

Reply:
Fixed in latest.